impeller/entity/contents/tiled_texture_contents.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "impeller/entity/contents/path_contents.h"
+#include "impeller/entity/entity.h"
+#include "impeller/geometry/path.h"
+#include "impeller/renderer/sampler_descriptor.h"
+
+namespace impeller {
+
+class TiledTextureContents final : public PathContents {
+ public:
+  TiledTextureContents();
+
+  ~TiledTextureContents() override;
+
+  void SetPath(Path path) override;
+
+  // |Contents|
+  std::optional<Rect> GetCoverage(const Entity& entity) const override;
+
+  // |Contents|
+  bool Render(const ContentContext& renderer,
+              const Entity& entity,
+              RenderPass& pass) const override;
+
+  void SetTexture(std::shared_ptr<Texture> texture);
+
+  void SetTileModes(Entity::TileMode horizontal_tile_mode,
+                    Entity::TileMode vertical_tile_mode);
```

Comment:
Any reason we can't just add this new option to the existing TextureContents instead? We can always add more specialized shaders later when optimizing for common cases, but I don't think it would necessitate a separate command factory.

Reply:
If we add the new option to `TextureContents`, we probably also need to add a `Type` to `TextureContents` to identify whether it is a `kDefault` type or `kTiled` type，because the way they calculate `texture_coords` is very different: 

TextureContents
```
    const auto tess_result = Tessellator{}.Tessellate(
        path_.GetFillType(), path_.CreatePolyline(),
        [this, &vertex_builder, &coverage_rect, &texture_size](Point vtx) {
          VS::PerVertexData data;
          data.position = vtx;
          auto coverage_coords =
              (vtx - coverage_rect->origin) / coverage_rect->size;
          data.texture_coords =
              (source_rect_.origin + source_rect_.size * coverage_coords) /
              texture_size;
          vertex_builder.AppendVertex(data);
        });
```
TiledTextureContents (matrix is ​​not supported yet)
```
    const auto tess_result =
        Tessellator{}.Tessellate(path_.GetFillType(), path_.CreatePolyline(),
                                 [&vertex_builder, &texture_size](Point vtx) {
                                   VS::PerVertexData data;
                                   data.position = vtx;
                                   data.texture_coords = vtx / texture_size;
                                   vertex_builder.AppendVertex(data);
                                 });
```

Reply:
Interesting -- I see the other reason for the difference: The UVs don't get mapped to the coverage rect.